src/onelogin/saml2/response.py

```diff
@@ -549,3 +549,12 @@ def get_error(self):
         After executing a validation process, if it fails this method returns the cause
         """
         return self.__error
+
+    def get_xml_document(self):
+        """
+        If necessary, decrypt the XML response document, and return it.
+        """
```

Comment:
Docstring needs to specify the return type:
```
        :returns: XML document
        :rtype: Element
```

Also, now that this method exists, you could use it to simplify the `__query(self, query):` method just above.